threeplay.go

```diff
@@ -65,51 +84,81 @@ func NewClientWithHTTPClient(apiKey, apiSecret string, client HTTPClient) *Clien
 	}
 }
 
-const ThreePlayHost = "https://api.3playmedia.com"
+func (c Client) buildUrl(endpoint string, querystring url.Values) string {
+	querystring.Add("apikey", c.apiKey)
 
-type Error struct {
-	IsError bool              `json:"iserror"`
-	Errors  map[string]string `json:"errors"`
+	url := url.URL{
+		Scheme:   "https",
+		Host:     ThreePlayHost,
+		Path:     endpoint,
+		RawQuery: querystring.Encode(),
+	}
+
+	return url.String()
 }
 
-func (c *Client) GetFile(id uint) (*File, error) {
-	file := &File{}
+func (c Client) fetchAndParse(endpoint string, ref interface{}) error {
 	apiError := &Error{}
-	endpoint := fmt.Sprintf("%s/files/%d?apikey=%s", ThreePlayHost, id, c.apiKey)
 	response, err := c.client.Get(endpoint)
 
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	responseData, err := ioutil.ReadAll(response.Body)
 
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	err = json.Unmarshal(responseData, apiError)
 
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	if apiError.IsError {
-		return nil, errors.New("API Error")
+		return errors.New("API Error")
 	}
 
-	err = json.Unmarshal(responseData, file)
+	err = json.Unmarshal(responseData, ref)
 
 	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c Client) GetFiles(params ...url.Values) (*FilesPage, error) {
+	querystring := url.Values{}
+	if len(params) > 0 {
+		querystring = params[0]
+	}
+
+	filesPage := &FilesPage{}
+	endpoint := c.buildUrl("/files", querystring)
+	if err := c.fetchAndParse(endpoint, filesPage); err != nil {
+		return nil, err
+	} else {
```

Comment:
thats a good catch, I thought my `filesPage` were in the if scope and not outside of it
I'll setup these lints later